tracee-ebpf/tracee/tracee.bpf.c

```diff
@@ -2217,83 +2226,31 @@ int syscall__execveat(void *ctx)
     return events_perf_submit(&data, SYS_EXECVEAT, 0);
 }
 
-SEC("raw_tracepoint/sys_socket")
-int sys_socket_exit_tail(void *ctx)
+static __always_inline int check_fd_socket(u64 fd)
```

Comment:
You can make it a more generic function `check_fd_type(u64 fd, u16 type)` where you can pass the requested type as an argument (socket/fifo...)